client/apps.go

```diff
@@ -0,0 +1,88 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015-2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+)
+
+type AppInfo struct {
+	Snap         string `json:"snap,omitempty"`
+	Name         string `json:"name"`
+	DesktopFile  string `json:"desktop-file,omitempty"`
+	*ServiceInfo `json:",omitempty"`
+}
+
+// IsService returns true if the application is a background daemon.
+func (a *AppInfo) IsService() bool {
+	if a == nil {
+		return false
+	}
+	if a.ServiceInfo == nil {
+		return false
+	}
+	if a.ServiceInfo.Daemon == "" {
+		return false
+	}
+
+	return true
+}
+
+type ServiceInfo struct {
+	Daemon          string `json:"daemon"`
+	ServiceFileName string `json:"service-file-name"`
+	Enabled         bool   `json:"enabled"`
+	Active          bool   `json:"active"`
+}
+
+// AppInfoWanted can be used to specify what kind of app is of
+// interest.
+type AppInfoWanted struct {
+	Services bool
+	Commands bool
+}
+
+// AppInfos returns information about all matching apps. Each name can
+// be either a snap or a snap.app. If names is empty, list all.
+//
+// Use wanted to include apps of the given kind; nil means all.
+func (client *Client) AppInfos(names []string, wanted *AppInfoWanted) ([]*AppInfo, error) {
+	q := make(url.Values)
+	if len(names) > 0 {
+		q.Add("apps", strings.Join(names, ","))
+	}
+	if wanted != nil && !(wanted.Services && wanted.Commands) {
+		if !wanted.Services && !wanted.Commands {
+			return nil, errors.New("cannot pass in an empty AppInfoWanted")
+		}
+		wantedStr := "services"
+		if wanted.Commands {
+			wantedStr = "commands"
+		}
+		q.Add("wanted", wantedStr)
```

Comment:
You might be on to something there.
We've used `"select"` for something similar but not quite the same:

In `/v2/find` it can be either empty (with no way to explicitly specify whatever that behaviour is), `"private"` (which makes find return only private snaps), or `"refresh"` (which makes find return only locally installed snaps that have a refresh available). This is the oldest use case of `"select"`, so it's the one I think of, but also so clunky and slightly wrong that I'm immediately repulsed. However, there are more uses:

In `/v2/snaps` it can be `"enabled"` (the default), so it only lists enabled snaps, or `"all"`, which makes list return all installed snaps (see `snap list --all`).

In `/v2/changes` it can be either be `"in-progress"` (the default), or `"ready"`, or `"all"`.

So it looks like it being `"all"` or `"services"` would fit. If this is surfaced as a boolean in an options struct the asymmetry of that still puts me off a bit, but :man_shrugging: